src/StepManager.cpp

```diff
@@ -114,12 +115,34 @@ MatchResult StepManager::stepMatches(const std::string &stepDescription) {
         const boost::shared_ptr<const StepInfo>& stepInfo = iter->second;
         SingleStepMatch currentMatch = stepInfo->matches(stepDescription);
         if (currentMatch) {
+            for (auto& submatch : currentMatch.submatches) {
+                submatch.position = codePointsFromStart(stepDescription, submatch.position);
+            }
             matchResult.addMatch(currentMatch);
         }
     }
     return matchResult;
 }
 
+
+std::ptrdiff_t StepManager::codePointsFromStart(const std::string &expression, size_t index) {
+    std::ptrdiff_t codePointCount(0);
+    for (const uint8_t i : expression) {
+        if (i < 0x80) {
+            codePointCount++;
+        }
+        if ((i & 0b1100'0000) == 0b1100'0000) {
```

Comment:
https://en.cppreference.com/w/cpp/language/integer_literal

- 0b (binary literals) are >= c++14
- digit separators also are

So you'll have to convert it to an hexadecimal litteral, unfortunately